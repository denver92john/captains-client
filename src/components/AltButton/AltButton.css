.altbutton {}

.fa-altbutton-icon {
    font-size: 25px;
}

.altbutton-text {
    display: inline-block;
}