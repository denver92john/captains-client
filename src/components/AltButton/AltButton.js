import React from 'react';
import {Link} from 'react-router-dom';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
//import './AltButton.css';

function AltButton(props) {
    return (
        <div>
            <Link to={props.altTo}>
                <p className="altbutton-text">{props.altWhat}</p>
                <FontAwesomeIcon icon="user-friends" className="fa-altbutton-icon" />
            </Link>
        </div>
    );
}

export default AltButton;