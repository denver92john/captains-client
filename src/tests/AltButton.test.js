import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import AltButton from '../components/AltButton/AltButton';

describe('AltButton component', () => {
    it('renders without errors', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <AltButton altTo={'/teams'} altWhat={'Pick List'} isList={true} />
            </BrowserRouter>,
            div);
        ReactDOM.unmountComponentAtNode(div);
    });
    
    it('renders UI as expected', () => {
        const tree = renderer
            .create(<BrowserRouter>
                    <AltButton altTo={'/teams'} altWhat={'Pick List'} isList={true} />
                </BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})