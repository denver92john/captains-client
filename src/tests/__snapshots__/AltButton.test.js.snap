// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AltButton component renders UI as expected 1`] = `
<div>
  <a
    href="/teams"
    onClick={[Function]}
  >
    <p
      className="altbutton-text"
    >
      Pick List
    </p>
  </a>
</div>
`;
